File: Ejercicios Propuestos 1/Ejercicio 7/Ejercicio 7.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# El siguiente programa genera un número aleatorio comprendido entre 1 y el valor de la constante MAXIMO y posteriormente reta al usuario a que lo adivine. Para ello utiliza la función "random" que genera un número aleatorio x donde 0 ≤ x < 1.

# Importamos la librería random para generar números aleatorios
import random

# Generamos un número aleatorio entre 1 y 100
numero_secreto = random.randint(1, 100)

# Inicializamos el contador de intentos
contador = 0

# Variable donde se almacenará cada intento del usuario
intento = 0

# Mostramos mensaje de bienvenida
print("¡Bienvenido al juego de adivinar el número!")

# Repetimos mientras el usuario no adivine el número
while intento != numero_secreto:

    # Solicitamos un intento al usuario
    intento = int(input("Ingresa tu intento (entre 1 y 100): "))

    # Incrementamos el contador de intentos
    contador += 1
    
    # Verificamos si el intento es mayor que el número secreto
    if intento > numero_secreto:
        print("El número es menor")

    # Verificamos si el intento es menor que el número secreto
    elif intento < numero_secreto:
        print("El número es mayor")

    # Si no es mayor ni menor, entonces acertó
    else:
        print("¡Acertaste!")

# Mostramos la cantidad total de intentos realizados
print(f"Número de intentos: {contador}")

File: Ejercicios Propuestos 1/Ejercicio 7/Ejercicio7.java
import java.util.Scanner;
public class Ejercicio7 {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {
            final int MAXIMO = 100;
            int numeroSecreto = (int) (Math.random() * MAXIMO) + 1;
            boolean acierto = false;
            int intentos = 0; 
            System.out.println("He generado un número entre 1 y " + MAXIMO + ". ¡Intenta adivinarlo!");
            while (!acierto) {
                System.out.print("Introduzca un número: ");
                if (sc.hasNextInt()) {
                    int n = sc.nextInt();
                    intentos++;
                    if (n == numeroSecreto) {
                        acierto = true;
                        System.out.println("¡¡acertaste!!");
                    } else {
                        if (n < numeroSecreto) {
                            System.out.println("El número es mayor"); 
                        } else {
                            System.out.println("El número es menor");
                        }
                    }
                } else {
                    System.out.println("Por favor, introduce un número válido.");
                    sc.next();
                }
            }
            System.out.println("Has necesitado " + intentos + " intentos.");
        }
    }
}